addons/web_editor/controllers/main.py

```diff
@@ -402,52 +454,57 @@ def save_scss(self, url, bundle_xmlid, content):
             # If it was already modified, simply override the corresponding attachment content
             custom_attachment.write({"datas": datas})
         else:
-            # If not, create a new attachment to copy the original scss file content, with its modifications
+            # If not, create a new attachment to copy the original scss/js file content, with its modifications
             new_attach = {
                 'name': custom_url,
                 'type': "binary",
-                'mimetype': "text/scss",
+                'mimetype': (file_type == 'js' and 'text/javascript' or 'text/scss'),
                 'datas': datas,
                 'datas_fname': url.split("/")[-1],
                 'url': custom_url,
             }
-            new_attach.update(self.save_scss_attachment_hook())
+            new_attach.update(self.save_scss_or_js_attachment_hook())
             IrAttachment.create(new_attach)
 
             # Create a view to extend the template which adds the original file to link the new modified version instead
             IrUiView = request.env["ir.ui.view"]
             view_to_xpath = IrUiView.get_related_views(bundle_xmlid, bundles=True).filtered(lambda v: v.arch.find(url) >= 0)
+            tag = 'link' if file_type == 'scss' else 'script'
+            attribute = 'href' if file_type == 'scss' else 'src'
```

Comment:
Those two expressions could be directly in the dict below (no need of a temporary variable)